### PR TITLE
Add missing infra documentation

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR adds pieces of infra documentation that hasn't been moved over to this public repo. This is mostly FAQ entries and a DB restore playbook

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204610208832639/1204691860847698/f) _(internal)_

## 🔍 What does this change?

- Adjust README of Terraform folder
- Refactor another module in the config (container cluster)

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] requires changes to docs (It's a docs change PR)

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] does not affect the execution graph

## 🛡 What tests cover this?

- CI/CD would test the Terraform changes

## ❓ How to test this?

Check the Apply actions result or run `terraform plan --var-file prod-usea1.tfvars --refresh=false` from `/infra/terraform/hash`

### CODE REVIEW

File: infra/terraform/README.md [hunk 1]
```diff
@@ -21,35 +21,60 @@ prod
...
+1.  Migrate databases
+1.  Build/push docker images
 
-kratos_api_key = "secret"
+## Deploy infrastructure with terraform
 
-pg_superuser_password    = "changeme"
-pg_kratos_user_password  = { raw = "changeme", hash = "See guide below to generate this" }
-pg_graph_user_password   = { raw = "changeme", hash = "See guide below to generate this" }
+Secret environment should be provided in Hashicorp Vault. These are expected in a kvv2 path starting with `pipelines/hash/` and ending with the environment name, e.g. `automation/pipelines/hash/prod`. The following secrets are expected:
```
Comment (TimDiekmann) [inline infra/terraform/README.md:54]: Just a clarification question: With "expected" do you mean these are the required variables for deployment?

File: infra/terraform/README.md [hunk 2]
```diff
@@ -80,14 +105,14 @@ Deployment can then be done by issuing the following command after initializing
...
+$ terraform apply --var-file prod-usea1.tfvars
 ..
 ```
 
-Note that it may be required to disable refreshing state for subsequent applies (because of the Postgres SSH tunnel. Data sources are deferred to the apply phase usually).
+Note that it may be required to disable refreshing state for subsequent applies (because of the Postgres SSH tunnel. Data sources are deferred to the apply phase, usually).
 
 ```console
-$ terraform apply --var-file prod-usea1.tfvars --var-file prod.secrets.tfvars -refresh=false
+$ terraform apply --var-file prod-usea1.tfvars -refresh=false
```
Comment (TimDiekmann) [inline infra/terraform/README.md:115]:
```suggestion
$ terraform apply --var-file prod-usea1.tfvars --refresh=false
```

File: infra/terraform/README.md [hunk 3]
```diff
@@ -110,9 +135,7 @@ $ ./ssh_bastion.sh -N -L 5554:h-hash-dev-usea1-pg.*.us-east-1.rds.amazonaws.com:
 
 This will start an SSH tunnel making `localhost:5554` point to the remote RDS instance within the private subnet in AWS.
 
-To migrate the graph, you must first build the docker container that contains the graph and run it with the graph credentials and the you put into the [`./hash/prod.secrets.tfvars`](./hash/prod.secrets.tfvars) file using the `migrate` subcommand:
-
-Look at the next step and build `hash-graph`.
+To migrate the graph, you must first build the docker container that contains the graph and run it with the graph credentials you should have in the Vault instance.
```
Comment (TimDiekmann) [inline infra/terraform/README.md:138]:
```suggestion
To migrate `hash-graph`, you must first build the docker container that contains the graph and run it with the graph credentials you should have in the Vault instance.
```

File: infra/terraform/README.md [hunk 4]
```diff
@@ -201,21 +224,97 @@ $ aws ecs execute-command --cluster h-hash-prod-usea1-ecs --task TASKID --contai
 
 where the `TASKID` is the ID of the Fargate service task. Note that this would start an `sh` session directly into your production container, and should be done with caution (if it ever should be done). It's recommended not to run commands against the production environment like this, and we may at some point restrict access to Fargate Exec.
 
+# Troubleshooting and guides
+
+## What to do if the HASH DB needs **Disaster Recovery**
+
+There's a [playbook](./playbooks/db_recovery.md) on what we should do when the DB is on fire, please follow it to recover a snapshot.
```
Comment (TimDiekmann) [inline infra/terraform/README.md:231]:
```suggestion
There's a [recovery playbook](./playbooks/db_recovery.md) on what we should do when the DB is on fire, please follow it to recover a snapshot.
```

File: infra/terraform/README.md [hunk 5]
```diff
@@ -201,21 +224,97 @@ $ aws ecs execute-command --cluster h-hash-prod-usea1-ecs --task TASKID --contai
...
+
+## Help! I need access to the private subnet!
+
+You can access the private subnet using the Bastion host. See [this README](./hash/bastion.md) for more information. You need an SSH privatekey to access the bastion, which you can find in `1password` if you have the appropriate access level.
```
Comment (TimDiekmann) [inline infra/terraform/README.md:235]:
```suggestion
You can access the private subnet using the Bastion host. For more information, please see [the corresponding readme](./hash/bastion.md). You need an SSH privatekey to access the bastion, which you can find in `1password` if you have the appropriate access level.
```

File: infra/terraform/README.md [hunk 6]
```diff
@@ -201,21 +224,97 @@ $ aws ecs execute-command --cluster h-hash-prod-usea1-ecs --task TASKID --contai
...
+
+```console
+$ docker run --rm \
+  --network host \
+  -e HASH_GRAPH_PG_USER \
+  -e HASH_GRAPH_PG_PASSWORD \
+  -e HASH_GRAPH_PG_HOST \
+  -e HASH_GRAPH_PG_PORT \
+  -e HASH_GRAPH_PG_DATABASE \
+  -e RUST_LOG \
```
Comment (TimDiekmann) [inline infra/terraform/README.md:266]: Is it not required to actually pass the environment variables?

File: infra/terraform/README.md [hunk 7]
```diff
@@ -201,21 +224,97 @@ $ aws ecs execute-command --cluster h-hash-prod-usea1-ecs --task TASKID --contai
...
-- [`container_registry`](./modules/container_registry/) - is a thin wrapper around the `aws_ecr_repository` resource with container evicition policies
-- [`container_cluster`](./modules/container_cluster/) - is a thin wrapper around the `aws_ecs_cluster` resource.
+- [`container_cluster`](./modules/container_cluster/) - is a thin wrapper around the `aws_ecs_cluster` resource with container capacity providers.
+- [`container_registry`](./modules/container_registry/) - is a thin wrapper around the `aws_ecr_repository` resource with container evicition policies.
 - [`redis`](./modules/redis/) - [external service] a multi Availability Zone Redis cluster with encryption enabled.
+- [`tunnel`](./modules/tunnel/) - a custom SSH tunnel using an external data source to allow terraform to connect to services on the private subnet of the VPC.
+- [`variables`](./modules/variables/) - contains global variable validation/definitions generally useful for our Terraform infrastructure.
+- [`vault_aws_auth`](./modules/vault_aws_auth/) - contains the configuration for authenticating the AWS provider through the Vault AWS authentication backend.
+
+**HASH specefic modules**:
```
Comment (TimDiekmann) [inline infra/terraform/README.md:309]: Same as above, if there was a reason just resolve this comment
```suggestion
## HASH specefic modules
```

**Review by TimDiekmann — COMMENTED** [review]
Thanks @thehabbos007, the changes look good to me. I have a few questions, mainly for clarification.

File: infra/terraform/README.md (hunk 1 above)
Comment (thehabbos007) [reply infra/terraform/README.md]: Yes, the terraform config assumes/expects the path and the JSON structure to contain the secrets. We can change the wording if you want

File: infra/terraform/README.md (hunk 2 above)
Comment (thehabbos007) [reply infra/terraform/README.md:115]: Huh interesting that both `--` and `-` are valid. I'll apply the change for consistency

File: infra/terraform/README.md [hunk 8]
```diff
@@ -201,21 +224,97 @@ $ aws ecs execute-command --cluster h-hash-prod-usea1-ecs --task TASKID --contai
...
+Using the Terraform scripts to deploy an instance of RDS will give you an empty DB (with appropriate DBs/users), which will need to be migrated. From the `hashintel/hash` repo you can run the following command assuming you've set up an SSH tunnel (see [this README](./hash/bastion.md)) and have the appropriate environment variables set for the DB connection:
+
+```console
+$ docker run --rm \
+  --network host \
+  -e HASH_GRAPH_PG_USER \
+  -e HASH_GRAPH_PG_PASSWORD \
+  -e HASH_GRAPH_PG_HOST \
+  -e HASH_GRAPH_PG_PORT \
+  -e HASH_GRAPH_PG_DATABASE \
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/README.md:265]: They are, yes. The PG database one isn't though. It has to be read from the output of Terraform. Perhaps I should add explicit steps for setting these values

File: infra/terraform/README.md (hunk 1 above)
Comment (TimDiekmann) [reply infra/terraform/README.md]: I think the wording is good. I didn't realize this is actual JSON we expect here 🙂

File: infra/terraform/README.md (hunk 2 above)
Comment (TimDiekmann) [reply infra/terraform/README.md]: Oh, I thought it was a typo, didn't know that `-refresh` also works 😄

File: infra/terraform/README.md (hunk 1 above)
Comment (thehabbos007) [reply infra/terraform/README.md]: Oh i see. It is JSON, but the top level fields are treated as key-value pairs in vault. So the nested values are JSON objects, the top level ones are just vault key values :)

File: infra/terraform/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply infra/terraform/README.md:265]: I think this would be helpful, thank you!

File: infra/terraform/modules/citus/instance.tf [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+resource "aws_ebs_volume" "ebs1" {
+  availability_zone = var.az[var.region][0]
+  size              = 20 # GB
+  type              = "gp3"
+  tags = {
+    Name = "${local.prefix}-citusebs1"
+  }
+}
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/citus/instance.tf:156]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/citus/instance.tf [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+  subnet_id                   = data.terraform_remote_state.base.outputs.subnet_pub1_id
+  associate_public_ip_address = true
+  user_data                   = data.template_file.cloudinit.rendered
+  vpc_security_group_ids      = [aws_security_group.ssh.id]
+  iam_instance_profile        = aws_iam_instance_profile.instance.name
+  availability_zone           = var.az[var.region][0]
+  tags = {
+    Name = "${local.prefix}-citusec2"
+  }
+}
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/citus/instance.tf:147]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/opensearch/opensearch.tf [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+          "logs:PutLogEventsBatch",
+          "logs:CreateLogStream"
+        ],
+        Resource = ["arn:aws:logs:*"]
+      }
+    ]
+  })
+}
+
+resource "aws_elasticsearch_domain" "search" {
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/opensearch/opensearch.tf:49]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/citus/ecr.tf [hunk 12]
```diff
@@ -0,0 +1,4 @@
+resource "aws_ecr_repository" "citus" {
+  name = "${local.prefix}-citusecr"
+  tags = {}
+}
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/citus/ecr.tf:4]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/citus/instance.tf [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+
+resource "aws_ebs_volume" "ebs1" {
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/citus/instance.tf:149]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/citus/instance.tf (hunk 10 above)
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/citus/instance.tf:147]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/citus/ecr.tf (hunk 12 above)
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/citus/ecr.tf:4]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/citus/instance.tf (hunk 10 above)
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/citus/instance.tf:147]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/opensearch/opensearch.tf [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+# A Service Linked Role is required for AWS OpenSearch inside a VPC
+resource "aws_iam_service_linked_role" "search" {
+  aws_service_name = "es.amazonaws.com"
+}
+
+resource "aws_cloudwatch_log_group" "search" {
+  name              = "${local.prefix}-searchlog"
+  retention_in_days = 7
+  tags              = {}
+}
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/opensearch/opensearch.tf:26]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/citus/instance.tf (hunk 9 above)
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/citus/instance.tf:156]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/modules/citus/instance.tf (hunk 13 above)
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/modules/citus/instance.tf:149]: This is part of an archived module that is not in use. Any of these warnings should be addressed if we reuse the module.

File: infra/terraform/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (thehabbos007) [reply infra/terraform/README.md]: Addded in 4066e5e4cf66dd83d7cf7379d7db6ed706cfc11d

[the remark this answers is not in the document]
Comment (TimDiekmann) [reply infra/terraform/README.md]: Great, thank you!

**Review by TimDiekmann — APPROVED** [review]
Great work, looks good to me. I can imaging this being very useful in the future!

File: infra/terraform/playbooks/citus.md [hunk 15]
```diff
@@ -0,0 +1,94 @@
+# Citus
+
+Playbooks for interacting with the Citus database. Currently, we're running a
+single-node Citus setup on an EC2 instance.
```
Comment (nonparibus) [inline infra/terraform/playbooks/citus.md:4]: Do we want to rephrase this to make it clear that we're not actually running single-node Citus at the moment, but if we were using the Citus config _we would be_?

Comment (thehabbos007) [reply infra/terraform/playbooks/citus.md:4]:
```suggestion
This is a playbook for interacting with a self-hosted, single-node Citus cluster on EC2.
```

File: infra/terraform/README.md [hunk 16]
```diff
@@ -21,35 +21,60 @@ prod
...
-## Deploy infrastructure with terraform
+OWNER : 'h' for hash
+PROJECT : 'hash'
+ENV : 'dev' or 'prod'  (maybe others too later)
+REGION : AWS region shortened -- 'usea1' etc.
+RESOURCE DESCRIPTION : a short description of the resource e.g. 'subnetpub1'
+```
 
-You must prepare a `tfvars` file containing secrets to spin up the environment. This can be done by creating a file in `./hash/prod.secrets.tfvars` containing the following definitions (change the values):
+Fields may _not_ include any hyphens. Example names: `h-hash-prod-usea1-vpc`, `h-hash-dev-usea2-apisvc`.
```
Comment (CiaranMn) [inline infra/terraform/README.md:38]:
```suggestion
Fields (e.g. Owner, Project) may _not_ include any hyphens. 

Example _valid_ names: `h-hash-prod-usea1-vpc`, `h-hash-dev-usea2-apisvc`
Example _invalid_ name: `h-hash-prod-usea1-api-svc`
```

File: infra/terraform/README.md [hunk 17]
```diff
@@ -21,35 +21,60 @@ prod
...
 
-```tfvars
-hash_seed_users = [
-  { "email" = "admin@hash.ai", "shortname" = "instance-admin", "preferredName" = "Instance Admin", "password" = "changeme", "isInstanceAdmin" = true }
-]
+(Inspired by https://stepan.wtf/cloud-naming-convention/)
```
Comment (CiaranMn) [inline infra/terraform/README.md:40]:
```suggestion
(Inspired by [https://stepan.wtf/cloud-naming-convention](https://stepan.wtf/cloud-naming-convention/))
```

File: infra/terraform/README.md [hunk 18]
```diff
@@ -21,35 +21,60 @@ prod
...
+1.  Deploy infrastructure with terraform
+1.  Migrate databases
+1.  Build/push docker images
+
+## Deploy infrastructure with terraform
 
-pg_superuser_password    = "changeme"
-pg_kratos_user_password  = { raw = "changeme", hash = "See guide below to generate this" }
-pg_graph_user_password   = { raw = "changeme", hash = "See guide below to generate this" }
+Secret environment should be provided in HashiCorp Vault. These are expected in a kvv2 path starting with `pipelines/hash/` and ending with the environment name, e.g. `automation/pipelines/hash/prod`. The following secrets are expected:
```
Comment (CiaranMn) [inline infra/terraform/README.md:54]: where does `automation` come from here?

File: infra/terraform/README.md [hunk 19]
```diff
@@ -201,21 +224,128 @@ $ aws ecs execute-command --cluster h-hash-prod-usea1-ecs --task TASKID --contai
...
+## How do I log in to AWS ECR container registry?
+
+If you wish to pull/push container images to ECR, AWS CLI allows you to do so with the following command:
+
+```console
+$ aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin ACCOUNTID.dkr.ecr.us-east-1.amazonaws.com
+..
+```
+
+This will log into the given ECR in the `us-east-1` region.
```
Comment (CiaranMn) [inline infra/terraform/README.md:252]: This was missing steps in the process to 'pull/push container images to ECR', suggest we just link to the AWS docs which give them all
```suggestion
If you wish to pull/push container images to ECR manually, you must:
1. Login to the registry
1. Build and tag the image
1. Push the image to ECR
Please see the [AWS docs](https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html) for instructions on how to do so.
```

File: infra/terraform/playbooks/db_recovery.md [hunk 20]
```diff
@@ -0,0 +1,37 @@
+# HASH DB recovery playbook
+
+Something went terribly wrong with the database, here’s how to recover.
+
+## Identify snapshot to restore
+
+The HASH database is backed up every day in a 30-minute window in the interval 03:00–11:00 UTC.
+this means that we’ll have daily, incremental backups for every day in the snapshot history of the RDS instance.
+
```
Comment (CiaranMn) [inline infra/terraform/playbooks/db_recovery.md:9]: Not sure it's worth mentioning this since it might drift from reality and isn't important to the process
```suggestion
```

**Review by CiaranMn — COMMENTED** [review]
Thanks @thehabbos007, super helpful! A few non-blocking comments (I'm off next week so do what you will with them)

File: infra/terraform/README.md (hunk 18 above)
Comment (thehabbos007) [reply infra/terraform/README.md:57]: That's the engine name, I've tried to make it more clear in 82fe967394043693d34176331d985bef0c8d92ce
